File: src/main/java/yte/intern/project/application/usecases/ManageEvents/Entity/Participant.java
package yte.intern.project.application.usecases.ManageEvents.Entity;

import lombok.*;
import yte.intern.project.application.usecases.common.Entity.BaseEntity;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.List;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@SequenceGenerator(name="idgen",sequenceName = "EVENT_SEQ")
public class Participant extends BaseEntity {

    @Column( name = "NAME")
    private String name;

    @Column( name = "SURNAME")
    private String surname;

    @Column( name = "E_MAIL")
    private String email;

    @Column( name = "TC_KIMLIK_NO")
    private String tcKimlikNo;

    @Column( name = "JOIN_DATE")
    private LocalDate joinDate;

    @Column(name ="JOIN_TIME")
    private LocalTime joinTime;

    @Column(name="QR_CODE")
    private byte[] qrCode;

    @Column(name="AGE")
    private Long age;

    @Column(name="GENDER")
    private String gender;

    @OneToMany( cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "PARTICIPANT_ID")
    private Set<Answer> answers;

    @OneToMany( cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "PARTICIPANT_ID")
    private Set<SurveyQuestion> surveyQuestions;

    @OneToMany( cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "PARTICIPANT_ID")
    private Set<Chat> chats;



}

File: FlutterMobileExtension/lib/models/Participant.dart
import 'package:flutter/material.dart';
import 'package:tubitak_intern_mobile_project/models/Question.dart';
import 'package:tubitak_intern_mobile_project/models/SurveyQuestion.dart';



class Participant {
  String name;
  String surname;
  String email;
  String tcKimlikNo;
  String joinDate;
  String joinTime;
  int age;
  String gender;
  List<Question> answers;
  List<SurveyQuestion> surveyQuestions;

  Participant(
      {this.name,
      this.surname,
      this.email,
      this.tcKimlikNo,
      this.joinDate,
      this.joinTime,
      this.age,
      this.gender,
      this.answers,
      this.surveyQuestions});
}
